core/src/main/java/org/hisp/dhis/android/core/trackedentity/internal/TrackerQueryFactory.kt

```diff
@@ -51,7 +51,7 @@ internal abstract class TrackerQueryFactory<T, S : TrackerBaseSync>(
         val programSettings = programSettingsObjectRepository.blockingGet()
         val internalFactory = internalFactoryCreator.invoke(params, programSettings)
         lastUpdatedManager.prepare(programSettings, params)
-        return if (params.program() == null) {
+        return if (params.program() == null && params.programStageWorkingLists() == null) {
```

Comment:
I think this condition have to check `trackedEntityInstanceFilters` and `eventFilters` as well. Maybe it can be abstracted in a method in `params`.